native-schema-registry/c/src/memory_allocator.c

```diff
@@ -0,0 +1,21 @@
+#include "memory_allocator.h"
+#include "aws/common/allocator.h"
+
+/*
+ * Uses the default memory allocator from AWS SDK Common library to allocate / free memory.
+ */
+void *aws_common_malloc(size_t size) {
+    struct aws_allocator *allocator = aws_default_allocator();
```

Comment:
aws_default_allocator doesn't change. So you could store it at file scope to save on function calls if you'd like. Though, to be fair, you're allocating memory so a single fn call is unlkely to be noticeable.

Reply:
Are you suggestion to create a `static` member? We don't have a global initializer yet, so couldn't find a way to initialize it once.

Reply:
You can do it lazily which that function does under the hood anyways, or use AWS_thread_call_once